docs/howto/partitions.rst

```diff
@@ -0,0 +1,167 @@
+******************************************
+Add partition support to an application
+******************************************
+
+Partitions basics
+=================
+
+Applications implementing ``craft-application`` and its dependent suite of
+libraries can optionally make use of `disk partitions
+<https://en.wikipedia.org/wiki/Disk_partitioning>`_.  The supported list of
+partitions must be defined by the application itself (at runtime), and those
+defined partitions can then be used by application consumers' configuration
+yaml files.
+
+Optionally, partitions can be namespaced, for organizational purposes.
+
+Partition names and namespace names must consist of *only* lower-case
+alphabetic characters, unless a partition exists under a namespace, in which
+case it may also contain hyphen characters, though the first and last
+characters must still be alphabetic.
+
+``default`` must always be the first listed partition.
+
+In the below examples, we work with three partitions: ``default``, ``kernel``,
+and ``component/bar-baz``.  The ``default`` and ``kernel`` partitions do not
+have a namespace.  The ``bar-baz`` partition is part of the ``component``
+namespace.
+
+.. _app_changes:
+
+Required application changes
+============================
+
+To add partition support to an application, two basic changes are needed:
+
+#. Enable the feature
+
+   Use the :class:`Features <craft_parts.Features>` class to specify that the
+   application will use partitions:
+
+   .. code-block:: python
+
+     from craft_parts import Features
+
+     Features.reset()
+     Features(enable_partitions=True)
+
+   .. NOTE::
+      The ``craft-application`` class :class:`AppFeatures
+      <craft_application.AppFeatures>` has a similar name and serves a similar
+      purpose to ``craft-parts``'s :class:`Features <craft_parts.Features>`,
+      but partitions cannot be enabled via :class:`AppFeatures
+      <craft_application.AppFeatures>`!
+
+#. Define the list of partitions
+
+   We need to tell the :class:`LifecycleManager <craft_parts.LifecycleManager>`
+   class about our partitions, but applications do not usually directly
+   instantiate the LifecycleManager.
+
+   Instead, override your :class:`Application
+   <craft_application.Application>`'s ``_setup_partitions`` method, and return
+   a list of the partitions, which will eventually be passed to the
+   :class:`LifecycleManager <craft_parts.LifecycleManager>`:
+
+   .. code-block:: python
+
+     class SnackcraftApplication(Application):
+
+       ...
+
+       @override
+       def _setup_partitions(self, yaml_data: dict[str, Any]) -> list[str] | None:
+           return ["default", "kernel", "component/bar-baz"]
+
+Using the partitions
+====================
+
+Partitions cannot be used until `after you have configured your application
+<#app-changes>`_.
+
+In configuration yaml
```

Comment:
I think we should call this something else to avoid confusion - I suppose in general we refer to this as the "project file"

Reply:
I've been using the "project metadata file".  If you prefer "project file", I will start using that in my documentation.

Reply:
I've been using "project file", perhaps we should start a glossary/style guide with preferred terms to keep our communication consistent.

Reply:
Good stuff guys, will update.